File: day2/day2.js
import { input, testInput } from "./input.js";

const part1 = () => {
  console.log("Welcome To Day 2");
  const calulateWin = (n1, n2) => {
    let returnedTotal = 0;
    if (n2 === "X") {
      returnedTotal += 1;
    }
    if (n2 === "Z") {
      returnedTotal += 3;
    }
    if (n2 === "Y") {
      returnedTotal += 2;
    }

    // A || X = Rock and + 1
    // C || Z = Scisors and + 3
    // B || Y = Paper and + 2

    // Win = 6
    // Draw = 3
    // Loss = 0
    if (
      (n1 === "A" && n2 === "X") ||
      (n1 === "C" && n2 === "Z") ||
      (n1 === "B" && n2 === "Y")
    ) {
      returnedTotal += 3;
    }
    if (
      (n1 === "A" && n2 === "Y") ||
      (n1 === "C" && n2 === "X") ||
      (n1 === "B" && n2 === "Z")
    ) {
      returnedTotal += 6;
    }
    // console.log("returnedTotal", returnedTotal);
    return returnedTotal;
  };

  const splitInput = input.split("\n");

  let total = 0;
  splitInput.forEach((set) => {
    const splitSet = set.split(" ");
    total += calulateWin(splitSet[0], splitSet[1]);
  });

  console.log(total);
};

// part1();

const part2 = () => {
  console.log("Welcome To Day 2: Part 2");
  const calulateWin = (n1, n2) => {
    let returnedTotal = 0;

    if (n2 === "X") {
      if (n1 === "A") {
        returnedTotal += 3;
      }
      if (n1 === "B") {
        returnedTotal += 1;
      }
      if (n1 === "C") {
        returnedTotal += 2;
      }
    }
    if (n2 === "Y") {
      if (n1 === "A") {
        returnedTotal += 1;
      }
      if (n1 === "B") {
        returnedTotal += 2;
      }
      if (n1 === "C") {
        returnedTotal += 3;
      }
      returnedTotal += 3;
    }
    if (n2 === "Z") {
      if (n1 === "A") {
        returnedTotal += 2;
      }
      if (n1 === "B") {
        returnedTotal += 3;
      }
      if (n1 === "C") {
        returnedTotal += 1;
      }
      returnedTotal += 6;
    }
    // A || X = Rock and + 1
    // C || Z = Scisors and + 3
    // B || Y = Paper and + 2

    // Win = 6
    // Draw = 3
    // Loss = 0
    return returnedTotal;
  };

  const splitInput = input.split("\n");

  let total = 0;
  splitInput.forEach((set) => {
    const splitSet = set.split(" ");
    total += calulateWin(splitSet[0], splitSet[1]);
  });

  console.log(total);
};

part2();

// 12513

File: 2025/day2/day2.py
from input import input_data, test_input

def part1(data_source):
    print('Welcome To Day 2')
    # Split input into lines
    total = 0
    lines = data_source.strip().split(',')
    for line in lines:
        start, end = line.split('-')
        start = int(start)
        end = int(end)
        for num in range(start, end + 1):
            first_half = str(num)[:len(str(num))//2]
            second_half = str(num)[len(str(num))//2:]
            if first_half == second_half:
                total += num
    print(total)

def has_repeating_pattern(num):
    num_str = str(num)

    for pattern_length in range(1, len(num_str) // 2 + 1):
        pattern = num_str[:pattern_length]
        repeats = len(num_str) // pattern_length
        reconstructed = pattern * repeats
        if num_str == reconstructed:
            return True
    return False


if __name__ == '__main__':
    # part1(test_input)
    # part1(input_data)
    
    def part2(data_source):
        print('Welcome To Day 2: Part 2')
        lines = data_source.strip().split(',')
        total = 0
        for line in lines:
            start, end = line.split('-')
            start = int(start)
            end = int(end)
            for num in range(start, end + 1):
                if has_repeating_pattern(num):
                    total += num
        print(total)
        # Your solution here
        pass
    
    # part2(test_input)
    part2(input_data)
